File: emergency-server.js
const http = require('http');
const fs = require('fs');
const path = require('path');

const PORT = process.env.PORT || 3000;
const PUBLIC_DIR = path.join(__dirname, 'public');

const server = http.createServer((req, res) => {
  console.log(`📞 Request: ${req.method} ${req.url}`);
  
  let filePath;
  if (req.url === '/') {
    filePath = path.join(PUBLIC_DIR, 'index.html');
  } else {
    filePath = path.join(PUBLIC_DIR, req.url);
  }
  
  fs.readFile(filePath, (err, data) => {
    if (err) {
      console.log(`❌ Error loading ${filePath}:`, err.message);
      res.writeHead(404, { 'Content-Type': 'text/html' });
      res.end(`
        <html>
          <body style="font-family: Arial; padding: 20px; text-align: center; background: #f0f0f0;">
            <h1 style="color: red;">🚛 FleetFlow - File Not Found</h1>
            <p>Could not find: ${req.url}</p>
            <p><a href="/index.html">🏠 Go to Emergency Dashboard</a></p>
            <p><a href="/debug.html">🔧 Debug Page</a></p>
          </body>
        </html>
      `);
      return;
    }
    
    const ext = path.extname(filePath);
    let contentType = 'text/html';
    if (ext === '.css') contentType = 'text/css';
    if (ext === '.js') contentType = 'application/javascript';
    if (ext === '.json') contentType = 'application/json';
    
    res.writeHead(200, { 'Content-Type': contentType });
    res.end(data);
    console.log(`✅ Served: ${filePath}`);
  });
});

server.listen(PORT, () => {
  console.log('🚛 FleetFlow Emergency Server Started!');
  console.log(`📡 Server running at: http://localhost:${PORT}`);
  console.log(`📂 Serving from: ${PUBLIC_DIR}`);
  console.log('🌐 Available pages:');
  console.log('   • http://localhost:3000/ (Emergency Dashboard)');
  console.log('   • http://localhost:3000/debug.html (Debug Page)');
  console.log('   • http://localhost:3000/test.html (Test Page)');
});

server.on('error', (err) => {
  console.error('❌ Server error:', err.message);
  if (err.code === 'EADDRINUSE') {
    console.log('🔄 Port 3000 is busy, trying port 3001...');
    server.listen(3001);
  }
});

File: emergency-server.py
#!/usr/bin/env python3
import http.server
import socketserver
import os
import json
from urllib.parse import urlparse

PORT = 3000
DIRECTORY = "public"

class CustomHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=DIRECTORY, **kwargs)
    
    def do_GET(self):
        parsed_path = urlparse(self.path)
        path = parsed_path.path
        
        # Serve root as index.html
        if path == '/':
            self.path = '/index.html'
        
        return super().do_GET()

def main():
    # Change to the FleetFlow directory
    os.chdir('/Users/deedavis/FLEETFLOW')
    
    with socketserver.TCPServer(("", PORT), CustomHTTPRequestHandler) as httpd:
        print(f"🚛 FleetFlow Emergency Server starting on port {PORT}")
        print(f"📂 Serving from: {os.getcwd()}/public")
        print(f"🌐 Open: http://localhost:{PORT}")
        print("✅ Emergency dashboard available!")
        httpd.serve_forever()

if __name__ == "__main__":
    main()
